zigate/responses.py

```diff
@@ -715,26 +715,20 @@ class R804E(Response):
 
     def decode(self):
         Response.decode(self)
-        additionnal = self.data.pop('additionnal')
+        LOGGER.debug("R804E data=%s", self.data)
+        additional = self.data.pop('additional')
         neighbours = []
         for i in range(self.data['count']):
-            neighbour, additionnal = self._decode('!HQQBBB',
+            if len(additional) < 3:
```

Comment:
This avoids errors like these (zigate firmware 3.1a) :

```
2020-01-03 00:27:44 DEBUG (ZiGate-Decode data) [zigate] Received response 0x804e: b'38c10701060000'
2020-01-03 00:27:44 DEBUG (ZiGate-Decode data) [zigate] Dispatch ZIGATE_RESPONSE_RECEIVED
2020-01-03 00:27:44 DEBUG (ZiGate-Decode data) [zigate] R804E data=OrderedDict([('sequence', 56), ('status', 193), ('entries', 7), ('count', 1), ('index', 6), ('additionnal', b'\x00\x00'), ('lqi', 0)])
2020-01-03 00:27:44 ERROR (ZiGate-Decode data) [zigate] Error decoding response 0x804e: b'38c10701060000'
2020-01-03 00:27:44 ERROR (ZiGate-Decode data) [zigate] Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/zigate/core.py", line 595, in decode_data
    response = RESPONSES.get(msg_type, Response)(value, lqi)
  File "/usr/local/lib/python3.7/site-packages/zigate/responses.py", line 36, in __init__
    self.decode()
  File "/usr/local/lib/python3.7/site-packages/zigate/responses.py", line 726, in decode
    additionnal)
  File "/usr/local/lib/python3.7/site-packages/zigate/responses.py", line 109, in _decode
    sdata = OrderedDict(zip(keys, struct.unpack(fmt, data[:size])))
struct.error: unpack requires a buffer of 21 bytes
```

Reply:
could you add a test to check if the code is working with old and new firmware ?

Reply:
Well, I do not have response captures for older firmware versions, and I'm not sure this error is specific to version 3.1a, I just wanted to mentions all details I could think of.

I've added a test case for the response message `b'38c10701060000'`: https://github.com/doudz/zigate/pull/159/commits/523fea485cc0ca1ec71351ac531b9709cb81be69